src/check.c

```diff
@@ -166,10 +206,34 @@ static void tcase_free(TCase * tc)
     check_list_free(tc->ch_sflst);
     check_list_free(tc->unch_tflst);
     check_list_free(tc->ch_tflst);
-
+    tcase_set_tags(tc, NULL);
     free(tc);
 }
 
+unsigned int tcase_matching_tag(TCase *tc, List *check_for)
+{
+
+    if ((NULL == tc->tags) || (NULL == check_for))
+    {
+	return 0;
+    }
+    
+
+    for(check_list_front(check_for); !check_list_at_end(check_for);
+        check_list_advance(check_for))
+    {
+	for(check_list_front(tc->tags); !check_list_at_end(tc->tags);
+	    check_list_advance(tc->tags))
+	{
+	    if (0 == strcmp(check_list_val(tc->tags), check_list_val(check_for)))
```

Comment:
Clang did not like this:
```
16:15:47 check.c:228:15: fatal error: no matching function for call to 'strcmp'
16:15:47             if (0 == strcmp(check_list_val(tc->tags), check_list_val(check_for)))
16:15:47                      ^~~~~~
16:15:47 /usr/include/string.h:143:12: note: candidate function not viable: cannot convert argument of incomplete type 'void *' to 'const char *'
16:15:47 extern int strcmp (__const char *__s1, __const char *__s2)
16:15:47            ^
```
Might need to cast the contents of the list item to char*